homeassistant/components/upnp/config_flow.py

```diff
@@ -127,8 +127,18 @@ async def async_step_import(self, import_info: Optional[Mapping]):
             _LOGGER.info("No UPnP devices discovered, aborting")
             return self.async_abort(reason="no_devices_found")
 
-        discovery = self._discoveries[0]
-        return await self._async_create_entry_from_discovery(discovery)
+        # Ensure complete discovery.
+        discovery_info = self._discoveries[0]
+        if DISCOVERY_USN not in discovery_info:
+            _LOGGER.debug("Incomplete discovery, ignoring")
+            return self.async_abort(reason="incomplete_discovery")
+
+        # Ensure not already configuring/configured.
+        usn = discovery_info[DISCOVERY_USN]
+        await self.async_set_unique_id(usn)
+        self._abort_if_unique_id_configured()
```

Comment:
And this case.

Reply:
I think this prevents it from happening at all: https://github.com/home-assistant/core/blob/e482f6b11344b24b637f1bc6f7400f051603995d/homeassistant/components/upnp/config_flow.py#L114-L120

Do you think I can remove it? Then the flow will be aborted if a duplicate `unique_id` is encountered.

Reply:
We never pass any `import_info` to the import step, so that case cant happen either. It's not clear to me how the issue behavior was occurring.

Reply:
It prevents (re-)discovering in some cases, lets keep it.

Reply:
Test added.

Reply:
When does it prevent rediscovering?

Reply:
Hi @StevenLooman , will this bugfix be integrated in next version or I have to pull it manually?
I've the same problem with a continuosly discovered router.

Thanks

Reply:
@MartinHjelmare If `import_info` is never provided then I'll simply remove it. 

Rediscovery (the actual discovery of upnp-devices) was prevented in case `import_info` was given with the proper data, as it aborted the flow.

Reply:
@lucaorioli I am not sure. Please, also, do put any reactions/questions in the issue and not in this pull request.

Reply:
We can remove line 130 since we already know that there are no current entries.

Reply:
You are right. Removed.

Reply:
Bump. This line is still here.

Reply:
```suggestion
```

Reply:
Must have forgotten to push, sorry.